repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java

```diff
@@ -183,13 +184,19 @@ public void onChange(ChangedTypeDefs changedTypeDefs) throws AtlasBaseException
             if (CollectionUtils.isNotEmpty(changedTypeDefs.getCreatedTypeDefs())) {
                 for (AtlasBaseTypeDef typeDef : changedTypeDefs.getCreatedTypeDefs()) {
                     updateIndexForTypeDef(management, typeDef);
+                    if(typeDef.isIndexFailed()) {
+                        throw new IndexException("Index not created.");
```

Comment:
- please add logs
- include proper error code for this exception.
- Add more details in the message, like which index we are trying to access.